internal/services/admin/server.go

```diff
@@ -25,6 +25,19 @@ import (
 
 func (a *AdminServiceImpl) TriggerWorkflow(ctx context.Context, req *contracts.TriggerWorkflowRequest) (*contracts.TriggerWorkflowResponse, error) {
 	tenant := ctx.Value("tenant").(*dbsqlc.Tenant)
+
+	switch tenant.Version {
+	case dbsqlc.TenantMajorEngineVersionV0:
+		return a.triggerWorkflowV0(ctx, req)
+	case dbsqlc.TenantMajorEngineVersionV1:
+		return a.triggerWorkflowV1(ctx, req)
+	default:
+		return nil, status.Errorf(codes.Unimplemented, "TriggerWorkflow is not supported on major engine version %s", tenant.Version)
+	}
```

Comment:
qq - will the compiler fail here if we e.g. add a major engine version? just curious what the default behavior is in Go, in TS I know you need an explicit check like `const foo: never = xyz` to make sure you caught all the cases

Reply:
it won't fail to compile, no